elastic/kibana
# [TableListView] Enhance tag filtering

In this PR I've udpated the `<TableListView />` tag filtering mechanism. We can now filter by tags

* from the search bar filter dropdown
* directly from the table

Tags can be **included** or **excluded** (by Ctrl + click).

## Screenshots

<img width="1385" alt="Screenshot 2022-11-08 at 14 24 30" src="https://user-images.githubusercontent.com/2854616/200590200-0a89e636-0c18-4fcf-a290-ed17da60c36a.png">

<img width="1355" alt="Screenshot 2022-11-08 at 14 24 47" src="https://user-images.githubusercontent.com/2854616/200590214-d2fdaa28-3529-459b-9911-29934e89b305.png">

<img width="362" alt="Screenshot 2022-11-08 at 14 25 01" src="https://user-images.githubusercontent.com/2854616/200590216-beecc4fe-f642-4a66-9813-420451b9383e.png">

### How to test

* Navigate to the Dashboard / Visualisation library / Maps listing page
* Add some tags to the saved object if needed
* Filter by tag the table by 
  * Clicking the tag in the table or in the filter dropdown
  * Ctrl + click the tag in the table or in the filter dropdown
* Open the filter drop down and make sure the "Manage all tags" redirect correctly to the Tag management page  

## Plugin updated

The following plugins have been updated

* Dashboard
* Visualisation
* Maps

## Changes made to to the Saved object tagging API

* Added `getTagList()` handler to the api to allow consumers to retrieve the list of available tags.
* Added `tagRender()` handler to the `<TagList />` props to be able to control how the tag is rendered
* Updated the `parse_search_query` logic to check for `included` and `not included` tags and return a `tagReferencesToExclude` along with the `tagReferences` array.


## Release notes

The dashboard, visualize library and maps listing pages have been updated with an enhanced tag filtering. The new filtering allows tags to be either included or excluded from the search.

Fixes https://github.com/elastic/kibana/issues/135237
